diff --git a/flutter/lib/desktop/widgets/remote_toolbar.dart b/flutter/lib/desktop/widgets/remote_toolbar.dart
index 34879e5..b401b37 100644
--- a/flutter/lib/desktop/widgets/remote_toolbar.dart
+++ b/flutter/lib/desktop/widgets/remote_toolbar.dart
@@ -3346,6 +3346,7 @@ class _DraggableShowHideState extends State<_DraggableShowHide> {
       mainAxisSize: MainAxisSize.min,
       children: [
         _buildDraggable(context),
+        _CycleMonitorMenu(id: widget.id, ffi: widget.ffi),
         Obx(() => collapse.isTrue
             ? _MinimizedMonitorSwitchButton(id: widget.id, ffi: widget.ffi)
             : const Offstage()),
@@ -3579,3 +3580,50 @@ class _MinimizedMonitorSwitchButton extends StatelessWidget {
     });
   }
 }
+
+class _CycleMonitorMenu extends StatelessWidget {
+  final String id;
+  final FFI ffi;
+
+  const _CycleMonitorMenu({
+    Key? key,
+    required this.id,
+    required this.ffi,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final pi = ffi.ffiModel.pi;
+
+    return TextButton(
+      onPressed: () {
+        RxInt display = CurrentDisplayState.find(id);
+        display.value = display.value + 1;
+        if (display.value >= pi.displays.length) {
+          display.value = 0;
+        }
+        openMonitorInTheSameTab(display.value, ffi, pi);
+        pi.currentDisplay = display.value;
+      },
+      child: Stack(children: [
+        Container(
+            child: Align(
+                alignment: Alignment.center,
+                child: const Icon(
+                  Icons.personal_video,
+                  color: _ToolbarTheme.blueColor,
+                  size: 20.0,
+                ))),
+        Container(
+            child: Align(
+                alignment: const Alignment(0.0, -0.4),
+                child: Text(
+                  "  ${CurrentDisplayState.find(id).value + 1}/${pi.displays.length}",
+                  style: const TextStyle(
+                      color: _ToolbarTheme.blueColor, fontSize: 8),
+                ))),
+      ]),
+    );
+  }
+}
+
